### PR TITLE
Compute minimum storage required for the most memory intensive of evaluation.

### PR DESCRIPTION
**Pull request by bimalgaudel**

### CODE REVIEW

File: SeQuant/core/asy_cost.cpp [hunk 1]
```diff
@@ -202,6 +215,14 @@ bool operator>(AsyCost const &lhs, AsyCost const &rhs) {
   return !(lhs < rhs || lhs == rhs);
 }
 
+bool operator<=(AsyCost const& lhs, AsyCost const& rhs) {
+  return lhs < rhs || lhs == rhs;
```
Comment (Krzmbrzl) [inline SeQuant/core/asy_cost.cpp:219]:
```suggestion
  return !(lhs > rhs);
```
(tiny bit more efficient)

File: SeQuant/core/asy_cost.cpp [hunk 2]
```diff
@@ -202,6 +215,14 @@ bool operator>(AsyCost const &lhs, AsyCost const &rhs) {
...
+}
+
+bool operator>=(AsyCost const& lhs, AsyCost const& rhs) {
+  return lhs > rhs || lhs == rhs;
```
Comment (Krzmbrzl) [inline SeQuant/core/asy_cost.cpp:223]:
```suggestion
  return !(lhs < rhs);
```

File: SeQuant/core/binary_node.hpp [hunk 3]
```diff
@@ -9,6 +10,77 @@
...
+/// Visit children nodes followed by parent node.
+/// Left will be visited before right.
+struct PreOrder {};
+
+/// Visit parent node followed by children nodes.
+/// Left will be visited before right.
+struct PostOrder {};
+
+/// Visit left node then parent node followed by right node.
+struct InOrder {};
```
Comment (Krzmbrzl) [inline SeQuant/core/binary_node.hpp:27]: Might it make sense to group these into an enum?
```suggestion
enum class IterationOrder {
  Pre,
  Post,
  In,
};
```

Comment (bimalgaudel) [reply SeQuant/core/binary_node.hpp:27]: Then we cannot use tag dispatch. Which I intend to use.

Comment (Krzmbrzl) [reply SeQuant/core/binary_node.hpp:27]: Ah, I see.

One could `if constexpr` on the enum value and call the desired overloads, but I guess that could get clumsy :eyes:

**Pull request closed without merging**